Source/CBL_ForestDBStorage.mm

```diff
@@ -944,27 +944,52 @@ - (CBL_Revision*) addDocID: (NSString*)inDocID
         CBLStatusToOutNSError(*outStatus, outError);
         return nil;
     }
-
-    __block NSData* json = nil;
-    if (properties) {
-        json = [CBL_Revision asCanonicalJSON: properties error: NULL];
-        if (!json) {
-            *outStatus = kCBLStatusBadJSON;
-            CBLStatusToOutNSError(*outStatus, outError);
-            return nil;
-        }
-    } else {
-        json = [NSData dataWithBytes: "{}" length: 2];
-    }
-
+    
     __block CBL_Revision* putRev = nil;
     __block CBLDatabaseChange* change = nil;
 
     *outStatus = [self inTransaction: ^CBLStatus {
         NSString* docID = inDocID;
-        C4Slice prevRevIDSlice = revID2slice(inPrevRevID);
+        
+        if (inPrevRevID) { // Check if an existing doc?
```

Comment:
Also check whether `properties.cbl_attachments != nil` since otherwise there's nothing to do. That allows us to skip loading the doc from the db when there are no attachments.